functions/assignOfflineContact.ts

```diff
@@ -0,0 +1,141 @@
+import '@twilio-labs/serverless-runtime-types';
+import {
+  Context,
+  ServerlessCallback,
+  ServerlessFunctionSignature,
+} from '@twilio-labs/serverless-runtime-types/types';
+import {
+  responseWithCors,
+  bindResolve,
+  error400,
+  error500,
+  success,
+  send,
+} from '@tech-matters/serverless-helpers';
+// eslint-disable-next-line prettier/prettier
+import type { PromiseValue } from 'type-fest';
+
+const TokenValidator = require('twilio-flex-token-validator').functionValidator;
+
+type EnvVars = {
+  TWILIO_WORKSPACE_SID: string;
+  TWILIO_CHAT_TRANSFER_WORKFLOW_SID: string;
+};
+
+export type Body = {
+  targetSid?: string;
+  finalTaskAttributes: string;
+};
+
+// eslint-disable-next-line prettier/prettier
+type TaskInstance = PromiseValue<ReturnType<ReturnType<ReturnType<ReturnType<Context['getTwilioClient']>['taskrouter']['workspaces']>['tasks']>['fetch']>>;
+// eslint-disable-next-line prettier/prettier
+type WorkerInstance = PromiseValue<ReturnType<ReturnType<ReturnType<ReturnType<Context['getTwilioClient']>['taskrouter']['workspaces']>['workers']>['fetch']>>;
+
+const assignToAvailableWorker = async (
+  event: Body,
+  newTask: TaskInstance,
+) => {
+  const reservations = await newTask.reservations().list();
+  const reservation = reservations.find(r => r.workerSid === event.targetSid);
+
+  if (reservation) {
+    await reservation.update({ reservationStatus: 'accepted' });
+    await reservation.update({ reservationStatus: 'completed' });
+    return { type: 'success' } as const;
+  } 
+
+  await newTask.remove();
+  return {
+    type: 'error',
+    payload: { status: 500, message: 'Error: reservation for task not created.' },
+  } as const;
+  
+};
+
+const assignToOfflineWorker = async (
+  context: Context<EnvVars>,
+  event: Body,
+  targetWorker: WorkerInstance,
+  newTask: TaskInstance,
+) => {
+  const previousActivity = targetWorker.activitySid;
+  const previousAttributes = JSON.parse(targetWorker.attributes);
+  const availableActivity = await context
+    .getTwilioClient()
+    .taskrouter.workspaces(context.TWILIO_WORKSPACE_SID)
+    .activities.list({ friendlyName: 'Available' });
+
+  await targetWorker.update({
+    activitySid: availableActivity[0].sid,
+    attributes: JSON.stringify({ ...previousAttributes, waitingOfflineContact: true, acceptOnlyTask: newTask.sid }), // waitingOfflineContact & acceptOnlyTask are routing rules used to avoid other tasks to be assigned during this window of time
```

Comment:
> waitingOfflineContact & acceptOnlyTask are routing rules used to avoid other tasks

Do we need to add any TaskRouter rules to our documentation for setting up new environments?

Reply:
Yes! The additional expression is in the description of this PR, so we don't lose it. I've added it in dev environment only so far.

Reply:
Okay -- we'll have to make sure we remember to do that when deploying this to the different accounts